youtube_dl/extractor/msn.py

```diff
@@ -0,0 +1,80 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+from ..utils import int_or_none
+
+
+class MSNIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?msn\.com/.+/[a-z]{2}-(?P<id>[a-zA-Z]+)'
+    _TESTS = [{
+        'url': 'http://www.msn.com/en-ae/foodanddrink/joinourtable/criminal-minds-shemar-moore-shares-a-touching-goodbye-message/vp-BBqQYNE',
+        'md5': '8442f66c116cbab1ff7098f986983458',
+        'info_dict': {
+            'id': 'BBqQYNE',
+            'title': 'Criminal Minds - Shemar Moore Shares A Touching Goodbye Message',
+            'description': 'Following news of his departure from the show, the star of Criminal Minds shares a heartwarming message to his fans and "baby girls."',
+            'duration': '104',
+            'ext': 'mp4'
+        }
+    }, {
+        'url': 'http://www.msn.com/en-ae/news/offbeat/meet-the-nine-year-old-self-made-millionaire/ar-BBt6ZKf',
+        'md5': 'bfbb6edbc806d34218c7f326d3dddd74',
+        'info_dict': {
+            'id': 'BBt6ZKf',
+            'title': 'All That Bling: Self-Made Millionaire Child Builds Fashion &amp; Jewellery Empire',
+            'description': 'md5:8e683bd5c729d5fb16d96539a582aa5e',
+            'duration': '350',
+            'ext': 'mp4',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        self.report_extraction(video_id)
+        data = self._parse_json(self._search_regex(r'data-metadata\s*=\s*["\'](.+)["\']',
+            webpage, video_id, group=1).replace('&quot;', '"'), video_id)
+
+        subtitles = {}
+        try:
+            for file in data['files']:
+                if file['formatCode'] == '3100':
+                    lang = file['culture']
+                    subtitles[lang] = []
+                    subtitles[lang].append({
+                        'ext': 'ttml',
+                        'url': file['url'].replace('&amp;', '&'),
+                    })
+        except KeyError:
```

Comment:
Avoid too wide try sections, if any of the files fails it won't process the rest of them.